lib/rust/ensogl/core/src/display/shape/primitive/system.rs

```diff
@@ -909,3 +910,52 @@ macro_rules! _shape {
         }
     };
 }
+
+/// Defines a new cached shape system.
+///
+/// The outcome is the same as for [`shape!`] macro, but the shape will be near application start
+/// to the special "cached shapes" texture. The texture is available in GLSL as "pass_cached_shapes"
+/// uniform. In the future there will be also a possibility of parametrization of normal shapes by
+/// cached shapes (see [#184212663](https://www.pivotaltracker.com/story/show/184212663)).
+///
+/// Because shape, once cached, is not redrawn, we don't allow for any parameterization, even by
+/// styles.
```

Comment:
Wait, we should allow styles, right? Right now, we supporto nly one style (precompiled shaders bake styles on startup), but in the future, if style changes, the shape can be re-drawn to the texture. Styles could be cool, for some other elements than icons - we can just use colors from styles to draw shapes without the need to re-color them later.